resources/idlharness.js

```diff
@@ -3288,15 +3292,21 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                const validations = WebIDL2.validate(astArray);
+                if (validations.length) {
+                    const message = validations.map(v => v.message).join("\n\n");
+                    setup_error = setup_error || new Error(message);
```

Comment:
https://wpt.fyi/results/?diff&filter=ADC&q=seq%28status%3Apass%20status%3Afail%29&run_id=290840005&run_id=288800001 shows that quite many tests will start failing the setup step now, and most of them because of changes that are being made to Web IDL that take time to trickle through to all specs, like requiring `[Exposed=Window]`.

To not mask other problem, can we wrap this in a subtest called "idl_test validation", even if it's done within the setup code?

Reply:
That sounds good, could you tell me how I can add a subtest here? I'm not very familiar with idlharness yet 🤯

Reply:
Wrapping this bit of code in `test(() => { ... }, "IDL validation");` should work.